File: Deepspace_Java/P1/WeaponType.java
/**
 * @author BOLÍVAR PELÁEZ, CLARA
 * @author RUIZ GÓMEZ, SOLEDAD
 * 2º DGIIM - PDOO - Practice 1 DeepSpace
 */
package deepspace;

/**
 * @title WeaponType
 * @brief It represents the types of weapons
 */
public enum WeaponType { 
    
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f); 
    
    private float power; //DUDAAAAAAAAAAAAAAAAAAAA
    
    /**
    * @brief Constructor
    */
    WeaponType (float p) {
        this.power = p;
    }
    
    /**
    * @brief Getter with package visibility
    * @return power value
    */
    float getPower () {
        return power;
    }

}

File: Deepspace_Ruby/P1/WeaponType.rb
# BOLÍVAR PELÁEZ, CLARA
# RUIZ GÓMEZ, SOLEDAD
# 2º DGIIM - PDOO - Practice 1 DeepSpace

# WeaponType
# It represents the types of weapons

module DeepSpace
    module WeaponType
        
        class Type 

            # Constructor
            def initialize (p, n)
                @power=p
                @name=n
            end

            # Getter
            def power
                @power
            end

            # To string
            def to_s
                return "#{@name}"
            end
        end
        
        LASER   = Type.new(2.0, "LASER")
        MISSILE = Type.new(3.0, "MISSILE")
        PLASMA  = Type.new(4.0, "PLASMA")

    end
end